docs/guides/modules/test/pages/smarter-testing.adoc

```diff
@@ -41,565 +41,1571 @@ TIP: In codebases with sparse test coverage, Smarter Testing cannot accurately d
+
+TIP: Some test runners and language runtimes can have a reasonably large overhead getting to the point where they can start running tests, this interacts poorly with queue-based Dynamic Test Splitting.
+
+Add `options.dynamic-test-splitting: true` to your test suite configuration:
+[source,yaml]
+----
+# .circleci/test-suites.yml
+---
+name: ci tests
+discover: vitest list --filesOnly
+run: vitest run --reporter=junit --outputFile="<< outputs.junit >>" --bail 0 << test.atoms >>
+outputs:
+  junit: test-reports/tests.xml
+options:
+  dynamic-test-splitting: true
+----
+
+[#test-suite-configuration-options]
+== 4. Test suite configuration options
+
+The following options are available to be defined in the options map in config:
+
+[.table-scroll]
+--
+[cols=3*, options="header"]
+|===
+|Options Field|Default|Description
+
+|`dynamic-test-splitting`
+|false
+|Whether the tests should be distributed across a shared queue and fetched across multiple dynamic batches. +
+If a test runner has slow start up time per batch, disabling this can speed up tests.
+
+| `test-impact-analysis`
+|false
+|Enables the Smarter Testing features, such as test impact analysis.
+
+| `timeout`
+| 10
+| The time in minutes a step will wait for tests to become available when running in parallel.
+
+| `full-test-run-paths`
+a|
+* `.circleci/*.yml`
+* `go.mod`
+* `go.sum`
+* `package-lock.json`
+* `package.json`
+* `project.clj`
+* `yarn.lock`
+|A List of paths that might have an indirect impact on tests and should run the full test suite if a change is detected.
+To disable this option, provide an empty array. +
+`full-test-run-paths: []`
+
+|`test-analysis-duration`
+|null
+|The maximum duration test analysis will run for in minutes. +
+Any remaining tests will be analysed the next time test analysis is run.
+
+|`impact-key`
+|`"default"`
+|Group relevant impact data together using a matching key within the same project.
+|===
+--
+
+The following flags are available to be defined on the `circleci run testsuite` command.
+
+[#cli-flags]
+[.table-scroll]
+--
+[cols=3*, options="header"]
+|===
+|Flag|Default|Description
+
+|`--test-analysis=all\|impacted\|none`
+|On branch `main`, impacted. +
+On all other branches, `none`
+a| * `all` analyzes all discovered tests, used to override any existing impact data. +
+* `impacted` analyzes only tests impacted by a change, used to refresh impact data. +
+* `none` skips analysis.
+
+|`--test-selection=all\|impacted\|none`
+|On branch `main`, `all`. +
+On all other branches, `impacted`
+a| * `all` selects and runs all discovered tests, used to run the full test suite. +
+* `impacted` selects and runs only the tests impacted by a change. +
+* `none` skips running tests, used to skip straight to analysis.
+|===
+--
+
+== 5. Optional configuration
+
+=== 5.1 Use the `file-mapper` command when test atoms are *not* file names
```

Comment:
We should probably mention this step earlier. Is it possible to go through local onboarding without it?